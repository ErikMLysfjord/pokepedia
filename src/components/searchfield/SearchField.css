.searchfield {
  height: 4em;
  margin-bottom: 20px;
  background-color: #fff;
  color: #000000;
  border: 0;
  border-radius: 5px 0px 0px 5px;
  padding: 0 10px;
  width: 100%;
  outline: none;
  font-size: 16px;
}

.searchfield:focus {
}

.searchfield__container {
  margin-top: 1.5em;
  display: flex;
  gap: 0;
  width: 100%;
}

.searchfield__button {
  width: 4em;
  height: 4em;
  border-radius: 0px 5px 5px 0px;
  background-color: #db4032;
  border: 0;
  display: flex;
  justify-content: center;
  align-items: center;
  cursor: pointer;
  transition: 200ms;
}

.searchfield__button:hover {
  background-color: hsl(5, 70%, 60%);
}

.searchfield__image {
  height: 50%;
}
