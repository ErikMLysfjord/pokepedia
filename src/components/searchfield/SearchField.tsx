import { useState } from "react";
import "./SearchField.css";
import { useNavigate } from "react-router-dom";
import SearchIcon from "../../assets/search.png";

const SearchField = () => {
  const [searchText, setSearchText] = useState("");
  const navigate = useNavigate();

  return (
    <div className="searchfield__container">
      <input
        type="search"
        placeholder="Pokemon Name"
        className="searchfield"
        value={searchText}
        onChange={(e) => setSearchText(e.target.value)}
      />
      <button
        className="searchfield__button"
        onClick={() => navigate("/pokemon/" + searchText)}
      >
        <img className="searchfield__image" src={SearchIcon} />
      </button>
    </div>
  );
};
export default SearchField;
