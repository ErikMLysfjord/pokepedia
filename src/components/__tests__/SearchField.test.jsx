import { render, screen } from "@testing-library/react";
import { describe, expect, test } from "vitest";
import { fireEvent } from "@testing-library/dom";
import SearchField from "../searchfield/SearchField";

const mockUseNavigate = vi.fn();

vi.mock("react-router-dom", async () => {
  const actual = await vi.importActual("react-router-dom");
  return {
    ...actual,
    useNavigate: () => mockUseNavigate,
  };
});

describe("SearchField", () => {
  test("should render searchfield", () => {
    const { container } = render(<SearchField />);
    expect(container).toMatchSnapshot();
    expect(screen.getByRole("searchfield")).toBeInTheDocument();
    expect(screen.getByRole("searchfield")).toHaveValue("");
    expect(screen.getByRole("button")).toBeInTheDocument();
  });

  test("should change textfield value", () => {
    render(<SearchField />);
    const search = screen.getByRole("searchfield");
    fireEvent.change(search, { target: { value: "pikachu" } });
    expect(search.toHaveValue("pikachu"));
  });

  test("Pokemon page for the searched pokemon is shown on search submit", () => {
    render(<SearchField />);
    const search = screen.getByRole("searchfield");
    fireEvent.change(search, { target: { value: "vulpix" } });
    screen.getByRole("button").click;
    expect(mockUseNavigate).toHaveBeenCalledWith("/pokemon/vulpix");
  });

  test("Case sensitive search", () => {
    mockUseNavigate.mockClear();
    render(<SearchField />);
    const search = screen.getByRole("searchfield");
    fireEvent.change(search, { target: { value: "Vulpix" } });
    screen.getByRole("button").click;
    expect(mockUseNavigate).toHaveBeenCalledWith("/pokemon/vulpix");
  });
});
