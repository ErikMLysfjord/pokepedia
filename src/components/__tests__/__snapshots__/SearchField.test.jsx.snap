// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SearchField > should render searchfield 1`] = `
<div>
  <div
    class="searchfield__container"
  >
    <input
      class="searchfield"
      placeholder="Pokemon Name"
      role="searchfield"
      type="search"
      value=""
    />
    <button
      class="searchfield__button"
    >
      <img
        class="searchfield__image"
        src="/project1/src/assets/search.png"
      />
    </button>
  </div>
</div>
`;
